### PR TITLE
Frexp extensions

### PR DESCRIPTION
**Pull request by jtlap**
ifrexp replaces the old frexp
frexp now return two floating values

### CODE REVIEW

File: include/boost/simd/arch/common/scalar/function/rem_pio2.hpp [hunk 1]
```diff
@@ -422,7 +422,7 @@ do {                                                                           \
     }
 
     /* compute n */
-    z  = bs::ldexp(z,q0);        /* actual value of z */
+    z  = bs::pedantic_(ldexp)(z,q0);        /* actual value of z */
```
Comment (jfalcou) [inline include/boost/simd/arch/common/scalar/function/rem_pio2.hpp:425]: do we need pedantic here ?

File: include/boost/simd/arch/common/simd/function/expocvt.hpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+//   void writebits(bs::pack<T> a)
+//   {
+//     using iT =  bd::as_integer_t<T, unsigned>;
+
+//     std::cout << "("; writebits(bs::bitwise_cast<iT>(a[0]));
+//     std::cout << ",  ";
+//     writebits(bs::bitwise_cast<iT>(a[1]));
+//     std::cout << ")" ;
+//   }
+
```
Comment (jfalcou) [inline include/boost/simd/arch/common/simd/function/expocvt.hpp]: remove don't comment

File: test/function/simd/abs.regular.cpp [hunk 3]
```diff
@@ -82,3 +82,4 @@ STF_CASE_TPL( "Check abs behavior with signed integral", STF_SIGNED_INTEGRAL_TYP
   STF_EQUAL(abs(bs::One<p_t>()),  bs::One<r_t>());
   STF_EQUAL(abs(bs::Mone<p_t>()), bs::One<r_t>());
 }
+
```
Comment (jfalcou) [inline test/function/simd/abs.regular.cpp]: noise commit

File: test/function/simd/ldexp.pedantic.cpp [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+#include <boost/simd/constant/two.hpp>
+#include <boost/simd/constant/four.hpp>
+#include <boost/simd/constant/halfeps.hpp>
+#include <boost/simd/constant/three.hpp>
+#include <boost/simd/detail/constant/maxexponent.hpp>
+#include <boost/simd/detail/constant/limitexponent.hpp>
+#include <boost/simd/detail/constant/minexponent.hpp>
+#include <boost/simd/constant/smallestposval.hpp>
+#include <boost/simd/function/scalar/dec.hpp>
+#include <string>
```
Comment (jfalcou) [inline test/function/simd/ldexp.pedantic.cpp]: why string include ?

File: test/function/simd/ldexp.regular.cpp [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+#include <boost/simd/constant/two.hpp>
+#include <boost/simd/constant/four.hpp>
+#include <boost/simd/constant/halfeps.hpp>
+#include <boost/simd/constant/three.hpp>
+#include <boost/simd/detail/constant/maxexponent.hpp>
+#include <boost/simd/detail/constant/limitexponent.hpp>
+#include <boost/simd/detail/constant/minexponent.hpp>
+#include <boost/simd/constant/smallestposval.hpp>
+#include <boost/simd/function/scalar/dec.hpp>
+#include <string>
```
Comment (jfalcou) [inline test/function/simd/ldexp.regular.cpp]: again why string ?

File: test/function/simd/ldexp.cpp [hunk 6]
```diff
@@ -28,6 +30,7 @@
 #include <boost/simd/detail/constant/minexponent.hpp>
 #include <boost/simd/constant/smallestposval.hpp>
 #include <boost/simd/function/scalar/dec.hpp>
+#include <string>
```
Comment (jfalcou) [inline test/function/simd/ldexp.cpp]: again why string ?

File: test/function/simd/toint.saturated.cpp [hunk 7]
```diff
@@ -97,6 +97,7 @@ STF_CASE_TPL (" bs::saturated_(bs::toint) unsigned_int",  STF_UNSIGNED_INTEGRAL_
 {
   namespace bs = boost::simd;
   namespace bd = boost::dispatch;
+
```
Comment (jfalcou) [inline test/function/simd/toint.saturated.cpp:100]: noisy commit

File: include/boost/simd/arch/common/scalar/function/rem_pio2.hpp (hunk 1 above)
Comment (jtlap) [reply include/boost/simd/arch/common/scalar/function/rem_pio2.hpp:425]: Yes

File: include/boost/simd/arch/common/simd/function/expocvt.hpp (hunk 2 above)
Comment (jtlap) [reply include/boost/simd/arch/common/simd/function/expocvt.hpp]: expocvt will be removed in in following pull request, but ok I amend

File: test/function/simd/abs.regular.cpp (hunk 3 above)
Comment (jtlap) [reply test/function/simd/abs.regular.cpp]: ok

File: test/function/simd/ldexp.regular.cpp (hunk 5 above)
Comment (jtlap) [reply test/function/simd/ldexp.regular.cpp]: noise

File: test/function/simd/ldexp.pedantic.cpp (hunk 4 above)
Comment (jtlap) [reply test/function/simd/ldexp.pedantic.cpp]: idem

File: test/function/simd/ldexp.cpp (hunk 6 above)
Comment (jtlap) [reply test/function/simd/ldexp.cpp]: idem

File: test/function/simd/toint.saturated.cpp (hunk 7 above)
Comment (jtlap) [reply test/function/simd/toint.saturated.cpp:100]: ok

**Pull request merged**